### DISCUSSION TITLE
자식클래스를 컴포넌트를 갖는 GameObject를 외부클래스에서 GetComponent로 부모클래스를 가져올 경우 null을 반환합니다.

### DISCUSSION THREAD

**Discussion by YSHFM** [naverz/zepeto-studio-kor#858] [5. Q&A]
현재 시험삼아 동적풀을 구현해보고 있습니다.

EffectPoolManager라는 클래스에서
![image](https://user-images.githubusercontent.com/106574541/172754085-fbc57968-4207-4a09-a619-fff7b68ba94c.png)
이러한 코드를 실행했을때 EffectBase가 null로 반환됩니다.

예를들어 

![image](https://user-images.githubusercontent.com/106574541/172761087-dafecbdb-7eab-41ac-97ec-e24c06124940.png)
해당 클래스가 부모클래스이고

![image](https://user-images.githubusercontent.com/106574541/172754341-ea03ff67-c064-4cad-af9b-1fb8a23690b1.png)
![image](https://user-images.githubusercontent.com/106574541/172760988-92781ab6-f0a0-427d-baf6-f03ddfa3348e.png)

오브젝트가 Effect_Base를 상속받는 BustParticleEffect클래스를 갖고 있을 때,  EffectPoolManager의 위 코드를 실행할 경우 

![image](https://user-images.githubusercontent.com/106574541/172760661-64785405-2ff0-429a-a307-2a5c8f62ec88.png)
GetComponent한 결과가 null로 반환됩니다.

기존 유니티 C#코드에선 예전에 저렇게 했으면 자식클래스를 컴포넌트를 붙이더라도 정상적으로 가져왔던 걸로 기억하고 있는데 typescript는 다른가요? 아니면 내부 버그 이슈로 봐야될까요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, getComponent null버그입니다.
내부에서 수정 진행하였고 다음 마일스톤에 반영될 예정입니다.
감사합니다.

**Reply by ChrysoMilo** [reply]
다음 마일 스톤이 반영 예정 시기가 언제일까요??

**Reply by Zepeto-TechEvangelist** [reply]
일정 조율 중이어서 현재로서는 답변이 어렵습니다.
